test/e2e/nfd_topology_updater.go

```diff
@@ -23,97 +23,119 @@ import (
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
 
-		})
+		ginkgo.By("Waiting for daemonset pods to be ready")
+		gomega.Expect(e2epod.WaitForPodsReady(f.ClientSet, f.Namespace.Name, topologyUpdaterDaemonSet.Spec.Template.Labels["name"], 5)).NotTo(gomega.HaveOccurred())
+
+		label := labels.SelectorFromSet(map[string]string{"name": topologyUpdaterDaemonSet.Spec.Template.Labels["name"]})
+		pods, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).List(context.TODO(), metav1.ListOptions{LabelSelector: label.String()})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(pods.Items).ToNot(gomega.BeEmpty())
 
-		ginkgo.AfterEach(func() {
-			// Delete topology updater daemon set
-			err := f.ClientSet.AppsV1().DaemonSets(f.Namespace.Name).Delete(context.TODO(), topologyUpdaterDaemonSet.Name, metav1.DeleteOptions{})
-			if err != nil {
-				e2elog.Logf("failed to delete node topology updater daemon set: %v", err)
-			}
-
-			err = f.ClientSet.CoreV1().Services(f.Namespace.Name).Delete(context.TODO(), masterService.Name, metav1.DeleteOptions{})
-			if err != nil {
-				e2elog.Logf("failed to delete master service: %v", err)
-			}
-
-			err = f.ClientSet.CoreV1().Pods(f.Namespace.Name).Delete(context.TODO(), masterPod.Name, metav1.DeleteOptions{})
-			if err != nil {
-				e2elog.Logf("failed to delete master pod: %v", err)
-			}
-
-			err = testutils.DeconfigureRBAC(f.ClientSet, f.Namespace.Name)
-			if err != nil {
-				e2elog.Logf("failed to delete RBAC resources: %v", err)
-			}
-
-			err = extClient.ApiextensionsV1().CustomResourceDefinitions().Delete(context.TODO(), crd.Name, metav1.DeleteOptions{})
-			if err != nil {
-				e2elog.Logf("failed to delete node resources topologies CRD: %v", err)
-			}
+		topologyUpdaterNode, err = f.ClientSet.CoreV1().Nodes().Get(context.TODO(), pods.Items[0].Spec.NodeName, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		kubeletConfig, err = e2ekubelet.GetCurrentKubeletConfig(topologyUpdaterNode.Name, "", true)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	})
+
+	ginkgo.Context("with single nfd-master pod", func() {
+		ginkgo.It("should fill the node resource topologies CR with the data", func() {
+			gomega.Eventually(func() bool {
+				// TODO: we should avoid to use hardcoded namespace name
+				nodeTopology, err := topologyClient.TopologyV1alpha1().NodeResourceTopologies("default").Get(context.TODO(), topologyUpdaterNode.Name, metav1.GetOptions{})
+				if err != nil {
+					framework.Logf("failed to get the node topology resource: %v", err)
+					return false
+				}
+
+				if nodeTopology == nil || len(nodeTopology.TopologyPolicy) == 0 {
+					framework.Logf("failed to get topology policy from the node topology resource")
+					return false
+				}
+
+				if nodeTopology.TopologyPolicy[0] != (*kubeletConfig).TopologyManagerPolicy {
+					return false
+				}
+
+				// TODO: add more checks like checking distances, NUMA node and allocated CPUs
```

Comment:
I think that unless we provide somehow from the outside the values we need to expect (e.g. the test must be given the expected topology) we should only check we have SOME data which we need to unversally expect, like cpus >= 1
and len(numaZones) >= 1

Reply:
...Or maybe we can do some safe checks comparing with the node.status?

Reply:
Anyway: stuff for a followup PR.

Reply:
I thought to use your ghw adaptation to get resources, once it will be ready.